nifi-nar-bundles/nifi-ranger-bundle/nifi-ranger-plugin/src/main/java/org/apache/nifi/ranger/authorization/RangerNiFiAuthorizer.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.ranger.plugin.policyengine.RangerAccessRequestImpl;
+import org.apache.ranger.plugin.policyengine.RangerAccessResourceImpl;
+import org.apache.ranger.plugin.policyengine.RangerAccessResult;
+import org.apache.ranger.plugin.policyengine.RangerAccessResultProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.Date;
+
+/**
+ * Authorizer implementation that uses Apache Ranger to make authorization decisions.
+ */
+public class RangerNiFiAuthorizer implements Authorizer {
+
+    private static final Logger logger = LoggerFactory.getLogger(RangerNiFiAuthorizer.class);
+
+    static final String RANGER_AUDIT_PATH_PROP = "Ranger Audit Config Path";
+    static final String RANGER_SECURITY_PATH_PROP = "Ranger Security Config Path";
+    static final String RANGER_KERBEROS_ENABLED_PROP = "Ranger Kerberos Enabled";
+    static final String RANGER_ADMIN_IDENTITY_PROP = "Ranger Admin Identity";
+    static final String RANGER_SERVICE_TYPE_PROP = "Ranger Service Type";
+    static final String RANGER_APP_ID_PROP = "Ranger Application Id";
+
+    static final String RANGER_NIFI_RESOURCE_NAME = "nifi-resource";
+    static final String DEFAULT_SERVICE_TYPE = "nifi";
+    static final String DEFAULT_APP_ID = "nifi";
+    static final String RESOURCES_RESOURCE = "/resources";
+    static final String HADOOP_SECURITY_AUTHENTICATION = "hadoop.security.authentication";
+    static final String KERBEROS_AUTHENTICATION = "kerberos";
+
+    private volatile RangerBasePluginWithPolicies nifiPlugin = null;
+    private volatile RangerDefaultAuditHandler defaultAuditHandler = null;
+    private volatile String rangerAdminIdentity = null;
+    private volatile boolean rangerKerberosEnabled = false;
+
+    @Override
+    public void initialize(AuthorizerInitializationContext initializationContext) throws AuthorizerCreationException {
+
+    }
+
+    @Override
+    public void onConfigured(AuthorizerConfigurationContext configurationContext) throws AuthorizerCreationException {
+        try {
+            if (nifiPlugin == null) {
+                logger.info("RangerNiFiAuthorizer(): initializing base plugin");
+
+                final PropertyValue securityConfigValue = configurationContext.getProperty(RANGER_SECURITY_PATH_PROP);
+                addRequiredResource(RANGER_SECURITY_PATH_PROP, securityConfigValue);
+
+                final PropertyValue auditConfigValue = configurationContext.getProperty(RANGER_AUDIT_PATH_PROP);
+                addRequiredResource(RANGER_AUDIT_PATH_PROP, auditConfigValue);
+
+                final String rangerKerberosEnabledValue = getConfigValue(configurationContext, RANGER_KERBEROS_ENABLED_PROP, Boolean.FALSE.toString());
+                rangerKerberosEnabled = rangerKerberosEnabledValue.equals(Boolean.TRUE.toString()) ? true : false;
+
+                if (rangerKerberosEnabled) {
```

Comment:
Could not see testing coverage in this area, especially in the case where kerberos is enabled but no kerberos properties (keytab/principal) exist.